janus_server/tests/integration_test.rs

```diff
@@ -0,0 +1,58 @@
+use janus_server::{
+    aggregator::aggregator_server, client::Client, message::TaskId, trace::install_subscriber,
+};
+use std::net::{IpAddr, Ipv4Addr, SocketAddr};
+use url::Url;
+
+fn endpoint_from_socket_addr(addr: &SocketAddr) -> Url {
+    assert!(addr.ip().is_loopback());
+    let mut endpoint: Url = "http://localhost".parse().unwrap();
+    endpoint.set_port(Some(addr.port())).unwrap();
+
+    endpoint
+}
+
+#[tokio::test]
+#[serial_test::serial]
+async fn create_client() {
+    install_subscriber().unwrap();
+
+    let task_id = TaskId::random();
+
+    let (leader_address, leader_server) = aggregator_server(
+        task_id,
+        SocketAddr::new(IpAddr::V4(Ipv4Addr::new(127, 0, 0, 1)), 0),
+    );
+    let leader_handle = tokio::spawn(async move { leader_server.await });
```

Comment:
nit: leader_server is already a future, so we can spawn it without an async block. (same below with helper_server)
```suggestion
    let leader_handle = tokio::spawn(leader_server);
```